cdap-data-fabric/src/main/java/co/cask/cdap/data/view/ViewAdmin.java

```diff
@@ -42,19 +42,26 @@ public ViewAdmin(ViewStore store, ExploreFacade explore, ExploreTableNaming nami
   }
 
   public boolean createOrUpdate(Id.Stream.View viewId, ViewSpecification spec) throws Exception {
-    if (store.exists(viewId)) {
-      explore.disableExploreStream(viewId.getStream(), naming.getTableName(viewId));
+    try {
+      ViewSpecification previousSpec = store.get(viewId);
+      if (previousSpec == null) {
+        throw new IllegalStateException("View exists, but no specification found");
+      }
+      explore.disableExploreStream(viewId.getStream(), previousSpec.getTableName());
```

Comment:
So for updating, we actually allow changing the explore table name as well? Is it a good?